onnxruntime/core/providers/cpu/reduction/reduction_ops.h

```diff
@@ -5,22 +5,27 @@
 #define CORE_PROVIDERS_CPU_REDUCTION_OPS_H
 
 #include "core/common/common.h"
+#include "core/common/optional.h"
 #include "core/framework/op_kernel.h"
 
 namespace onnxruntime {
 
 template <bool allow_multi_axes>
 class ReduceKernelBase {
  protected:
-  ReduceKernelBase(const OpKernelInfo& info) {
+  ReduceKernelBase(const OpKernelInfo& info, optional<int64_t> keepdims_override = {}) {
```

Comment:
Scott McKay
Shouldn't be necessary to use a unique_ptr for a single int64_t value. Could just take an int64_t*, and caller can pass the address of an int64_t stack variable if they want to override.

Sherlock Huang
Yes. Made it optional